lib/processor/redis.go

```diff
@@ -0,0 +1,290 @@
+
+		/*
+			// TODO
+			mRedisSAddCount      stats.GetCounter("redis.sadd.count"),
+			mRedisSAddRetry      stats.GetCounter("redis.sadd.retry"),
+			mRedisSAddFailed     stats.GetCounter("redis.sadd.failed.error"),
+			mRedisSAddSuccess    stats.GetCounter("redis.sadd.success"),
+			mRedisSAddLatency    stats.GetTimer("redis.sadd.latency"),
+
+			mRedisScardCount      stats.GetCounter("redis.scard.count"),
+			mRedisScardRetry      stats.GetCounter("redis.scard.retry"),
+			mRedisScardFailed     stats.GetCounter("redis.scard.error.failed"),
+			mRedisScardNotFound   stats.GetCounter("redis.scard.error.not_found"),
+			mRedisScardSuccess    stats.GetCounter("redis.scard.success"),
+			mRedisScardLatency    stats.GetTimer("redis.scard.latency"),
+		*/
+	}
+
+	//
+
+	//
+
+	r.interpolatedValue = text.ContainsFunctionVariables(r.valueBytes)
+
+	if r.operator, err = getRedisOperator(conf.Redis.Operator, r.key, r.valueBytes, r.client); err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+//------------------------------------------------------------------------------
+
+type redisOperator func(key string, value []byte, client *redis.Client) (string, error)
+
+func newSCardOperator(key *text.InterpolatedString, value []byte, client *redis.Client) redisOperator {
+	return func(key string, value []byte, client *redis.Client) (string, error) {
+
+		res, err := client.SCard(key).Result()
+		if err != nil {
+			return "", err
+		}
+
+		return strconv.FormatInt(int64(res), 10), err
+	}
+}
+
+func newSAddOperator(key *text.InterpolatedString, value []byte, client *redis.Client) redisOperator {
+	return func(key string, value []byte, client *redis.Client) (string, error) {
+
+		res, err := client.SAdd(key, value).Result()
+		if err != nil {
+			return "", err
+		}
+
+		return strconv.FormatInt(int64(res), 10), err
+	}
+}
+
+func getRedisOperator(opStr string, key *text.InterpolatedString, value []byte, client *redis.Client) (redisOperator, error) {
+
+	switch opStr {
+	case "sadd":
+		return newSAddOperator(key, value, client), nil
+	case "scard":
+		return newSCardOperator(key, value, client), nil
+	}
+	return nil, fmt.Errorf("operator not recognised: %v", opStr)
+}
+
+// ProcessMessage applies the processor to a message, either creating >0
+// resulting messages or a response to be sent back to the message source.
+func (r *Redis) ProcessMessage(msg types.Message) ([]types.Message, types.Response) {
+	r.mCount.Incr(1)
+	newMsg := msg.Copy()
+
+	valueBytes := r.valueBytes
+	if r.interpolatedValue {
+		valueBytes = text.ReplaceFunctionVariables(msg, valueBytes)
+	}
+
+	proc := func(index int, span opentracing.Span, part types.Part) error {
+		key := r.prefix + r.key.Get(message.Lock(newMsg, index))
+
+		res, err := r.operator(key, valueBytes, r.client)
+		if err != nil {
+			r.log.Debugf("Operator failed for key '%s': %v\n", key, err)
+			return err
+		}
+		//
+
+		part.Metadata().Set("redis_returned_value", res)
```

Comment:
It's more consistent with other processors if you set the contents of the message with the result. Most of the time a processor like this is going to be used within https://docs.benthos.dev/processors/#process_field or https://docs.benthos.dev/processors/#process_map which already gives you the ability to store the result within metadata or a particular JSON field.